src/ECS/HUD/HUD.cpp

```diff
@@ -87,6 +89,46 @@ bool HUD::removeGroup(unsigned int groupID) {
     return groups.erase(groupID) != 0;
 }
 
+Entity *
+HUD::newButton(const string &text, const HUD::btn_callback &callback, glm::vec2 pos, glm::vec2 size, Entity *parent) {
```

Comment:
cannot do it this way because of z-sorting, make abs. interactable component with a bunch of callbacks instead and sort it by its group first, hook all up to the hud's main receiver